src/mscorlib/corefx/System/Globalization/CalendarData.Unix.cs

```diff
@@ -1,47 +1,363 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
-using System;
-using System.Runtime.InteropServices;
-using System.Runtime.CompilerServices;
-using System.Diagnostics.Contracts;
+
 using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Runtime.InteropServices;
+using System.Text;
 
 namespace System.Globalization
 {
+    // needs to be kept in sync with CalendarDataType in System.Globalization.Native
+    internal enum CalendarDataType
+    {
+        Uninitialized = 0,
+        NativeName = 1,
+        MonthDay = 2,
+        ShortDates = 3,
+        LongDates = 4,
+        YearMonths = 5,
+        DayNames = 6,
+        AbbrevDayNames = 7,
+        MonthNames = 8,
+        AbbrevMonthNames = 9,
+        SuperShortDayNames = 10,
+        MonthGenitiveNames = 11,
+        AbbrevMonthGenitiveNames = 12,
+        EraNames = 13,
+        AbbrevEraNames = 14,
+    }
+
+    // needs to be kept in sync with CalendarDataResult in System.Globalization.Native
+    internal enum CalendarDataResult
+    {
+        Success = 0,
+        UnknownError = 1,
+        InsufficentBuffer = 2,
+    }
+
     internal partial class CalendarData
     {
         private bool LoadCalendarDataFromSystem(String localeName, CalendarId calendarId)
         {
-            // TODO: Implement this.
-            return false;
+            bool result = true;
+            result &= GetCalendarInfo(localeName, calendarId, CalendarDataType.NativeName, out this.sNativeName);
```

Comment:
We follow this pattern of loading stuff even if parts fail in Windows as well, IIRC.  I would rather we not change it as I think the early out would make things look worse.  Do you feel strongly here?